.github/workflows/docker-publish.yml

```diff
@@ -3,15 +3,16 @@ name: Create and publish a Docker image
 
 # Configures this workflow to run every time a change is pushed to the main branch.
 on:
-  push:
-    branches: main
+  # push:
+  #   branches : ['main', 'dev']
   pull_request:
     branches: 'main'
```

Comment:
Sorry for the back and forth, 
I think that with the changes suggested below, the best way to go is

```yaml
on:
  push:
    branches: 'main'
  pull_request:
    branches: 'main'  
```

this way any push to main, or any PR open to main will trigger and build the docker image

Reply:
Changed